### PR TITLE
Add reference snaps for vpu-umd-test and openvino

### PR DESCRIPTION
**Pull request by frenchwr**
This adds reference snaps for two NPU-enabled applications:

* `vpu-umd-test`: runs a suite of tests to validate the user space driver + compiler
* `openvino`: python package for AIML workloads, uses NPU plugin

For now both apps need to be installed in developer mode in order to access the NPU device on the host. If you are interested in seeing logs I will include links in [PEK-976](https://warthogs.atlassian.net/browse/PEK-976). 

Next steps:

* Run with proper snap confinement (either by developing a new snap interface or possibly with the existing `custom-device` interface)
* Test with `core24` base (current snaps use `core22`)
* Explore different methods (extensions, content snaps,etc) for shipping NPU software components to other snaps

[PEK-976]: https://warthogs.atlassian.net/browse/PEK-976?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: snaps/vpu-umd-test/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+mkdir -p models/add_abc
+curl -o models/add_abc/add_abc.xml https://raw.githubusercontent.com/openvinotoolkit/openvino/master/src/core/tests/models/ir/add_abc.xml
+touch models/add_abc/add_abc.bin
+curl -o basic.yaml https://raw.githubusercontent.com/intel/linux-npu-driver/v1.5.1/validation/umd-test/configs/basic.yaml
+```
+
+Finally run the application:
+
+```
+vpu-umd-test --config=basic.yaml
```
Comment (hector-cao) [inline snaps/vpu-umd-test/README.md:95]: I m curious if we need the prefix `vpu-umd-test.` here

**Review by hector-cao — COMMENTED** [review]
LGTM

File: snaps/vpu-umd-test/README.md (hunk 1 above)
Comment (frenchwr) [reply snaps/vpu-umd-test/README.md:95]: Normally yes but the exception is when the app name and snap name match. From the [docs](https://snapcraft.io/docs/snapcraft-app-and-service-metadata#appsapp-name):

> If app-name is the same as name, the program will be invoked as app-name. However, if they differ, the program will be exposed as snap-name.app-name.

Comment (hector-cao) [reply snaps/vpu-umd-test/README.md:95]: Thanks !

**Review by hector-cao — APPROVED** [review]
Great work ! LGTM

**Pull request merged**